src/smk/tool/geomark/tool-geomark.js

```diff
@@ -109,14 +109,24 @@ include.module( 'tool-geomark', [
                 }
             } 
 
+            this.defaultTemplineStyle = undefined;
+            this.defaultHintlineStyle = undefined;
             this.changedActive( function () {
                 if ( self.active ) {
+                    if (!this.defaultTemplineStyle) {
```

Comment:
The true "default" path styles (the blue colour and so on) doesn't seem to be exposed in global options; they only seem to contain custom values that have been set.

This is also dependent on the styles being set in the Markup tool first, though this could be rewritten to be more defensive if styles have not been set. 

This could also be optimized to scale better (should we want to set more styles) by pulling it out to a function.

Reply:
Yeah, we definitely need to be more defensive. What do you think about reading values from config if available, and falling back to reasonable hard coded defaults?

Reply:
When you suggest reading values from "config", do you mean smk-config.json? I'd expect to add the config to the Markup tool then. 

Alternately, the default values could go into a function that is visible to all tools. I'm a bit concerned about having CSS rules defined in JS code; maybe we could find a way to read values from a CSS file.

Reply:
Correcting myself: Path options look like CSS but aren't, so I'm less concerned about defining them in JS code. Leaflet path options are defined in https://leafletjs.com/reference.html#path.

Reply:
Oh, sorry for the lack of clarity. I'm thinking faster than I can type! I'm thinking hard coded default color for the regular markup tools, and reading the colors for the Geomark module from config. Perhaps a little overkill for the feature though. It could always be done as an enhancement at some point in the future.

Would it be possible to allow the geomark and markup modules to be 'greedy'? ie. Set the color of their choosing in the this.changedActive function and not worrying about restoring previous values?

Reply:
Thanks for clarifying. 

About reading Geomark styles from config ... I'm agreeing that may be overkill. We'd have to do some design and discussion on which style options we'd want to be able to set, and how to set them.

About being greedy ... I don't think the Markup tool could be greedy because it's not a panel and can't know whether it's active to override any previously set style options. That's why my implementation sets the default in Markup, and then Geomark (and any other future panel-based tools that draw shapes) would need to set and unset style options on active and inactive events.

Reply:
AH, good point! I forgot about that. Also, there is almost no code in the markup tool to modify.